### PR TITLE
416 450 471 update documentation

### LINKED ISSUE #416: Update unit-test guidelines to require independent testing
## Subject of the documentation

Now we have some tests where we test different cases sequentially inside one single function.
This is not ideal as sometimes previous tests might have an unexpected impact on the following tests.

Imagine we test a function named `func` we can build tests as follows:

```python
class TestFunc:
    def test_case1(self):
        ...

    def test_case2(self):
        ...
```

so that case1 and case2 are isolated.

In case we are just changing the values of the arguments, we can use [Parametrizing tests](https://docs.pytest.org/en/stable/example/parametrize.html) to remove redundant code.

We do not have to refactor all existing code, but we should update the guidelines so that new tests or modified tests should be refactored.

If the documentation request is approved, would you be willing to submit a PR? _(Help
can be provided if you need assistance submitting a PR)_

Yes

### PR DESCRIPTION
**Pull request by mathpluscode**
# Description

Update documentation regarding installation and contributing (pre-commit setup / testing).

Fixes #416 #450 #471 

## Type of change

What types of changes does your code introduce to DeepReg? _Put an `x` in the boxes that
apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [x] Documentation Update (fix or improvement on the documentation)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Other (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR.
If you're unsure about any of them, don't hesitate to ask. We're here to help! This is
simply a reminder of what we are going to look for before merging your code._

- [x] I have
      [installed pre-commit](https://deepreg.readthedocs.io/en/latest/contributing/setup.html)
      using `pre-commit install` and formatted all changed files. If you are not
      certain, run `pre-commit run --all-files`.
- [x] My commits' message styles matches
      [our requested structure](https://deepreg.readthedocs.io/en/latest/contributing/commit.html),
      e.g. `Issue #<issue number>: detailed message`.
- [x] I have updated the
      [change log file](https://github.com/DeepRegNet/DeepReg/blob/main/CHANGELOG.md)
      regarding my changes.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by mathpluscode** [comment]
I've updated the docs together regarding #416 #450 #471.

@YipengHu @ebonmati @NMontanaBrown could you please have a look ;) The main idea is to simplify the docs.

**Comment by mathpluscode** [comment]
https://github.com/DeepRegNet/DeepReg/community Actually, maybe we should add contributing guidelines in one single file. Need to check what other OS projects do.

File: docs/source/getting_started/install.rst [hunk 1]
```diff
@@ -1,76 +1,98 @@
...
+3. Install directly from PyPI release without cloning `DeepReg`_.
 
-`Anaconda <https://docs.anaconda.com/anaconda/install/>`__ /
-`Miniconda <https://docs.conda.io/en/latest/miniconda.html>`__
+Install via Conda
+-----------------
 
-DeepReg primarily supports and is regularly tested with Ubuntu and Debian Linux distributions.
+The recommended method is to install DeepReg in a dedicated virtual
+environment using `Anaconda`_ / `Miniconda`_ to avoid issues with other dependencies.
```
Comment (YipengHu) [inline docs/source/getting_started/install.rst:17]: "to avoid to avoid dependency issues."

File: docs/source/getting_started/install.rst [hunk 2]
```diff
@@ -1,76 +1,98 @@
...
+
+Please clone `DeepReg`_ first and install DeepReg following the instructions below.
```
Comment (YipengHu) [inline docs/source/getting_started/install.rst:19]: - Does the `conda create` actually install deepreg? if not, we need to add this;
- Need to add `git clone`
- Need to add `cd DeepReg`
- Is there a way to customise to another conda env name? I can think of multiple workflows that can utilise this, ideally, something like `conda env create -n deepreg0 -f environment_cpu.yml`. If not, add a comment explaining one needs to modify the yml for changing the default name 'deepreg'

Also consider:
"Clone DeepReg repository, create the pre-defined conda environment and install DeepReg following the instructions below. See official documentation for details regarding the conda environment.

File: docs/source/getting_started/install.rst [hunk 3]
```diff
@@ -135,11 +148,16 @@ Install the package directly
 
 **Note**
 
-All dependencies, APIs and command-line tools will be installed automatically via either installation method.
-However, the PyPI release currently does not ship with test data and demos.
-Running examples in this documentation may require downloading test data and changing default paths to user-installed packages with the PyPI release.
-These examples include those in the `Quick Start`_ and `DeepReg Demo`_.
-
+1. All dependencies, APIs and command-line tools will be installed automatically via either installation method.
```
Comment (YipengHu) [inline docs/source/getting_started/install.rst:151]: either -> each

File: docs/source/getting_started/install.rst [hunk 4]
```diff
@@ -135,11 +148,16 @@ Install the package directly
...
+   However, the PyPI release currently does not ship with test data and demos.
+   Running examples in this documentation may require downloading test data
+   and changing default paths to user-installed packages with the PyPI release.
+   These examples include those in the `Quick Start`_ and `DeepReg Demo`_.
+2. Only released versions of DeepReg are available via PyPI release.
+   Therefore it is different from the `latest (unreleased) version <https://github.com/DeepRegNet/DeepReg>`__ on GitHub.
```
Comment (YipengHu) [inline docs/source/getting_started/install.rst:157]: unreleased -> unstable?

**Review by YipengHu — CHANGES_REQUESTED** [review]

File: docs/source/getting_started/install.rst (hunk 1 above)
Comment (mathpluscode) [reply docs/source/getting_started/install.rst]: done

File: docs/source/getting_started/install.rst (hunk 3 above)
Comment (mathpluscode) [reply docs/source/getting_started/install.rst]: done

File: docs/source/getting_started/install.rst (hunk 4 above)
Comment (mathpluscode) [reply docs/source/getting_started/install.rst]: done

File: docs/source/getting_started/install.rst (hunk 2 above)
Comment (mathpluscode) [reply docs/source/getting_started/install.rst]: for the renaming, that's advanced? check conda docs for the renaming.
why people have multiple deepreg env :(

**Review by YipengHu — APPROVED** [review]

**Comment by mathpluscode** [comment]
@RemiDelaunay @zacbaum would you mind have a look ;) ?

File: docs/source/getting_started/install.rst [hunk 5]
```diff
@@ -1,76 +1,118 @@
...
 We also provide the docker file for building the docker image.
+Please clone `DeepReg repository`_ first:
+
+.. code:: bash
+
+    git clone git@github.com:DeepRegNet/DeepReg.git
+
+then install DeepReg following the instructions below.
 
 Install docker
```
Comment (zacbaum) [inline docs/source/getting_started/install.rst:117]: Typo in line 142:

docker run --name <container_name> --privileged=true -ti deepreg bash

should be 

docker run --name <container_name> --privileged=true -it deepreg bash

**Review by zacbaum — CHANGES_REQUESTED** [review]
Typo in `install.rst` line142:

`docker run --name <container_name> --privileged=true -ti deepreg bash`

should be

`docker run --name <container_name> --privileged=true -it deepreg bash`

**Comment by mathpluscode** [comment]
> Typo in `install.rst` line142:
> 
> 
> 
> `docker run --name <container_name> --privileged=true -ti deepreg bash`
> 
> 
> 
> should be
> 
> 
> 
> `docker run --name <container_name> --privileged=true -it deepreg bash`

It's equivalent ti or it, they are two flags i and t @zacbaum

**Comment by zacbaum** [comment]
> > Typo in `install.rst` line142:
> > `docker run --name <container_name> --privileged=true -ti deepreg bash`
> > should be
> > `docker run --name <container_name> --privileged=true -it deepreg bash`
> 
> It's equivalent ti or it, they are two flags i and t @zacbaum

Ok great! Just saw that it said -it on the link you had there and in the description above. Didn't check the docker docs.

**Review by zacbaum — APPROVED** [review]

**Pull request merged**